ga4gh/cli.py

```diff
@@ -1510,448 +1521,595 @@ def configtest_main(parser=None):
 ##############################################################################
 
 
-def getRawInput(displayString):
-    userResponse = raw_input(displayString)
-    return userResponse
+def getNameFromPath(filePath):
+    """
+    Returns the filename of the specified path without its extensions.
+    This is usually how we derive the default name for a given object.
+    """
+    if len(filePath) == 0:
+        raise ValueError("Cannot have empty path for name")
+    fileName = os.path.split(os.path.normpath(filePath))[1]
+    # We need to handle things like .fa.gz, so we can't use
+    # os.path.splitext
+    ret = fileName.split(".")[0]
+    assert ret != ""
+    return ret
 
 
-class AbstractRepoCommandRunner(object):
+def getRawInput(display):
+    """
+    Wrapper around raw_input; put into separate function so that it
+    can be easily mocked for tests.
+    """
+    return raw_input(display)
+
 
+class RepoManager(object):
+    """
+    Class that provide command line functionality to manage a
+    data repository.
+    """
     def __init__(self, args):
-        self.args = args
-        self.repoPath = args.repoPath
-        self.repoManager = repo_manager.RepoManager(self.repoPath)
-        if 'force' in args:
-            self.force = args.force
-
-    def confirmRun(self, func, deleteString):
-        if self.force:
+        self._args = args
+        self._repoPath = args.repoPath
+        self._repo = datarepo.SqlDataRepository(self._repoPath)
+
+    def _confirmDelete(self, objectType, name, func):
+        if self._args.force:
             func()
         else:
             displayString = (
-                "Are you sure you want to delete data in {}? "
-                "[y|N] ".format(deleteString))
+                "Are you sure you want to delete the {} '{}'? "
+                "[y|N] ".format(objectType, name))
             userResponse = getRawInput(displayString)
             if userResponse.strip() == 'y':
                 func()
             else:
                 print("Aborted")
 
+    def _updateRepo(self, func, *args, **kwargs):
+        """
+        Runs the specified function that updates the repo with the specified
+        arguments. This method ensures that all updates are transactional,
+        so that if any part of the update fails no changes are made to the
+        repo.
+        """
+        # TODO how do we make this properly transactional?
```

Comment:
It's a database, so it has mechanisms in place that make each step transactional and depending on journaling mode, "rollback-able".

If you wanted to be extra safe, add a key to the system table that gets flipped during open and end of commit.